src/master-playlist-controller.js

```diff
@@ -321,37 +306,85 @@ export default class MasterPlaylistController extends videojs.EventTarget {
     this.masterPlaylistLoader_.load();
   }
 
+  fillAudioTracks_() {
+    let master = this.master();
+    let mediaGroups = master.mediaGroups || {};
+
+    // force a default if we have none or we are not
+    // in html5 mode (the only mode to support more than one
+    // audio track)
+    if (!mediaGroups ||
+        !mediaGroups.AUDIO ||
+        Object.keys(mediaGroups.AUDIO).length === 0 ||
+        this.mediaSourceMode !== 'html5') {
+      // "main" audio group, track name "default"
+      mediaGroups = videojs.mergeOptions(mediaGroups, {AUDIO: {
+        main: {default: {default: true}}}
+      });
+    }
+
+    let tracks = {};
+
+    for (let mediaGroup in mediaGroups.AUDIO) {
+      for (let label in mediaGroups.AUDIO[mediaGroup]) {
+        let properties = mediaGroups.AUDIO[mediaGroup][label];
+
+        // if the track already exists add a new "location"
+        // since tracks in different mediaGroups are actually the same
+        // track with different locations
+        if (tracks[label]) {
+          tracks[label].addLoader(mediaGroup, properties.resolvedUri);
+          continue;
+        }
+
+        let track = new HlsAudioTrack(videojs.mergeOptions(properties, {
+          withCredentials: this.withCredential,
+          mediaGroup,
+          label
+        }));
+
+        tracks[label] = track;
+        this.audioTracks().push(track);
+      }
+    }
+  }
+
   load() {
     this.mainSegmentLoader_.load();
     if (this.audioPlaylistLoader_) {
       this.audioSegmentLoader_.load();
     }
   }
 
-  useAudio() {
+  activeAudioGroup() {
     let media = this.masterPlaylistLoader_.media();
-    let master = this.masterPlaylistLoader_.master;
+    let mediaGroup = 'main';
 
-    if (!media || !master) {
-      videojs.log.warn('useAudio() was called before playlist was loaded');
-      return;
+    if (media && media.attributes && media.attributes.AUDIO) {
+      mediaGroup = media.attributes.AUDIO;
     }
 
-    // We have been called but there is no audio track data so we only have the main one
-    // (that we know about). An example of this is when the source URL was a playlist
-    // manifest, not a master.
-    if (!media.attributes || !media.attributes.AUDIO ||
-        !master.mediaGroups || !master.mediaGroups.AUDIO) {
-      return;
-    }
-    let mediaGroupName = media.attributes.AUDIO;
+    return mediaGroup;
+  }
+
+  audioTracks() {
```

Comment:
This property seems like the sort of thing that is project-internal to me. Could we get away with just using `this.audioTracks_` directly?